mclagsyncd/mclaglink.cpp

```diff
@@ -401,276 +519,1390 @@ void MclagLink::setFdbEntry(char *msg, int msg_len)
+            m_mclag_domains[domain] = domainData;
+
+            //send config msg to iccpd
+            SWSS_LOG_DEBUG(" MCLAGSYNCD CFG Table Updates : domain_id:%d op_type:%d attrBmap:0x%x attrDelBmap:0x%x cfg_info.local_ip %s, peer_ip: %s peer_link:%s system_mac:%s session_timeout:%d keepalive_time:%d ",
+                    domain.domain_id, cfgOpType, attrBmap, attrDelBmap, cfg_info.local_ip, cfg_info.peer_ip, cfg_info.peer_ifname, m_system_mac.c_str(), cfg_info.session_timeout, cfg_info.keepalive_time);
+
+            //Entry not found previously and got created now - do add operation
+            if (!entryExists)
+            {
+                cfgOpType = MCLAG_CFG_OPER_ADD;
+
+                add_cfg_dependent_selectables = 1;
+            }
+            else //entry found
+            {
+                //entry found and one attribute is deleted
+                if ( attrDelBmap && (attrBmap == attrDelBmap) )
+                {
+                    cfgOpType = MCLAG_CFG_OPER_ATTR_DEL;
+                }
+                else //entry found and attribute are getting updated
+                {
+                    cfgOpType = MCLAG_CFG_OPER_UPDATE;
+                }
+
+            }
+        }
+        else
+        {
+            //Entry not found - error deletion
+            if (!entryExists)
+            {
+                SWSS_LOG_WARN("mclagsycnd to ICCPD, cfg processing ; Domain [%d] deletion - domain not found", domain.domain_id);
+                return;
+            }
+            else
+            {
+                cfgOpType = MCLAG_CFG_OPER_DEL;
+                SWSS_LOG_NOTICE(" Del dependent selectables from select ");
+                delDomainCfgDependentSelectables();
+                add_cfg_dependent_selectables = 0;
+                m_mclag_domains.erase(domain);
+                SWSS_LOG_NOTICE("mclagsycnd to ICCPD, cfg processing ; Domain [%d] deletion", domain.domain_id);
+            }
+        }
+
+        if (cfgOpType == MCLAG_CFG_OPER_NONE)
+        {
+            SWSS_LOG_NOTICE("mclagsycnd to ICCPD, cfg processing ; Domain [%d] op type not set", domain.domain_id);
+            return;
+        }
+
+
+        if (MCLAG_MAX_SEND_MSG_LEN - infor_len < (sizeof(struct mclag_domain_cfg_info)) )
+        {
+            cfg_msg_hdr = reinterpret_cast<mclag_msg_hdr_t *>(static_cast<void *>(infor_start));
+            cfg_msg_hdr->version = 1;
+            cfg_msg_hdr->msg_len = (unsigned short)infor_len;
+            cfg_msg_hdr->msg_type = MCLAG_SYNCD_MSG_TYPE_CFG_MCLAG_DOMAIN;
+
+            SWSS_LOG_DEBUG("mclagsycnd send msg to iccpd, msg_len =%d, msg_type =%d version=%d count : %d", cfg_msg_hdr->msg_len, cfg_msg_hdr->msg_type, cfg_msg_hdr->version, count);
+            write = ::write(getConnSocket(), infor_start, cfg_msg_hdr->msg_len);
+
+            if (write <= 0)
+            {
+                SWSS_LOG_ERROR("mclagsycnd to ICCPD, domain cfg send, buffer full; write to m_connection_socket failed");
```

Comment:
What is the recovery for buffer full? should it be retried again later? returned to caller with error status? The code seems to jut continue as if no error occurred?